### PR TITLE
feat(forge): optional verifier url

### LINKED ISSUE #2876: Sourcify - configurable URL
### Component

Forge

### Describe the feature you would like

I'm running a private network, and I'm also running Sourcify instance connected to it.

At the moment, Sourcify URL is hardcoded: https://github.com/foundry-rs/foundry/blob/master/cli/src/cmd/forge/verify/sourcify.rs#L14, so I'm unable to verify contract to my Sourcify instance via Forge.

Tnx for your help!

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rkrasiuk**
close #2876

### CODE REVIEW

File: cli/src/cmd/forge/verify/sourcify.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ impl VerificationProvider for SourcifyVerificationProvider {
                         SimpleCast::checksum_address(&args.address)?
                     );
                     let response = client
-                        .post(SOURCIFY_URL)
+                        .post(args.verifier.verifier_url.as_ref().unwrap_or(&SOURCIFY_URL.to_owned()))
```
Comment (mattsse) [inline cli/src/cmd/forge/verify/sourcify.rs:91]: I believe `.as_deref` should work
```suggestion
                        .post(args.verifier.verifier_url.as_deref().unwrap_or(SOURCIFY_URL))
```

File: cli/src/cmd/forge/verify/sourcify.rs [hunk 2]
```diff
@@ -125,7 +125,7 @@ impl VerificationProvider for SourcifyVerificationProvider {
                 async {
                     let url = format!(
                         "{}check-by-addresses?addresses={}&chainIds={}",
-                        SOURCIFY_URL,
+                        args.verifier.verifier_url.as_ref().unwrap_or(&SOURCIFY_URL.to_owned()),
```
Comment (mattsse) [inline cli/src/cmd/forge/verify/sourcify.rs:128]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

File: cli/src/cmd/forge/verify/sourcify.rs (hunk 1 above)
Comment (rkrasiuk) [reply cli/src/cmd/forge/verify/sourcify.rs]: fixed

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**